### PR TITLE
Better Teleports

### PR DESCRIPTION
**Pull request by Kurfursten**
Backend change, no CL

Thanks @DamianX

### CODE REVIEW

File: code/game/objects/items/weapons/storage/lockbox.dm [hunk 1]
```diff
@@ -513,24 +513,24 @@
 	return ..()
 
 /obj/item/weapon/storage/lockbox/advanced/proc/react()
-	var/list/turfs = new/list()
+	var/list/best_options = list()
+	var/list/backup_options = list()
+	var/turf/picked
 	for(var/turf/T in orange(6, loc))
-		if(istype(T,/turf/space))
+		if(T.x>world.maxx-6 || T.x<6 || T.y>world.maxy-6 || T.y<6) //Conditions we will NEVER accept: too close to edge
```
Comment (Exxion) [inline code/game/objects/items/weapons/storage/lockbox.dm]: We should really have a proc/macro for this, but that's for another PR

**Comment by DamianX** [comment]
Copypasting is quite gross. Why not put this into an /obj or /movable proc?

**Comment by Kurfursten** [comment]
Hmm, could do, I see no reason not to

**Comment by Kurfursten** [comment]
Okay, now it is at movable, tested again to be safe

**Review by ShiftyRail — APPROVED** [review]

**Pull request merged**